src/solace_ai_connector/common/messaging/solace_messaging.py

```diff
@@ -136,8 +140,16 @@ def connect(self):
             .build()
         )
 
-        # Blocking connect thread
-        self.messaging_service.connect()
+        try:
+            # Blocking connect thread
+            self.messaging_service.connect()
+        except KeyboardInterrupt:
+            log.error("Interrupt received. Disconnecting from messaging service.")
+            self.disconnect()
+            exit()
```

Comment:
You shouldn't exit the program from inside a component, there are still a lot other components that need to be cleaned up

Reply:
Interrupt signal should not be caught from inside a component, the logic for that is here https://github.com/SolaceDev/solace-ai-connector/blob/main/src/solace_ai_connector/solace_ai_connector.py#L99

Reply:
@cyrus2281 I moved the exception handler around the `app.run` in the` main.py`. 
Note that this error happens when the workflow is initialized by the `run` method.